tests/Parse/ParseQueryTest.php

```diff
@@ -2720,4 +2720,56 @@ public function testUnknownCondition()
             'unrecognized'  => 1
         ]);
     }
+
+    public function testEqualToWithSameKeyAndWithOthersConditionsReturnWrongResult()
+    {
+        $baz = new ParseObject('TestObject');
+        $baz->setArray('fooStack', [
+            [
+                'status' => 'baz'
+            ],
+            [
+                'status' => 'bar'
+            ]
+        ]);
+        $baz->save();
+
+        $bar = new ParseObject('TestObject');
+        $bar->setArray('fooStack', [
+            [
+                'status' => 'bar'
+            ]
+        ]);
+        $bar->save();
+
+        $qux = new ParseObject('TestObject');
+        $qux->setArray('fooStack', [
+            [
+                'status' => 'bar',
+            ],
+            [
+                'status' => 'qux'
+            ]
+        ]);
+        $qux->save();
+
+        $query = new ParseQuery('TestObject');
+        $query->notEqualTo('fooStack.status', 'baz');
+        $query->equalTo('fooStack.status', 'bar');
+
+        $this->assertEquals(3, $query->count(true));
```

Comment:
Is this the expected results? Should these test fail instead of passing?

Reply:
yes, they must fail. I just ran the tests, but after fixing it with the solution mentioned in issue #476 , they should fail